packages/web/hooks/use-measure.ts

```diff
@@ -0,0 +1,57 @@
+import { useMemo, useState } from "react";
+import { noop } from "../utils/function";
+import { isBrowser } from "../utils/ssr";
+import { useIsomorphicLayoutEffect } from "./use-isomorphic-layout-effect";
+
+export type UseMeasureRect = Pick<
+  DOMRectReadOnly,
+  "x" | "y" | "top" | "left" | "right" | "bottom" | "height" | "width"
+>;
+export type UseMeasureRef<E extends Element = Element> = (element: E) => void;
+export type UseMeasureResult<E extends Element = Element> = [
+  UseMeasureRef<E>,
+  UseMeasureRect
+];
+
+const defaultState: UseMeasureRect = {
+  x: 0,
+  y: 0,
+  width: 0,
+  height: 0,
+  top: 0,
+  left: 0,
+  bottom: 0,
+  right: 0,
+};
+
+function useMeasure<E extends Element = Element>(): UseMeasureResult<E> {
```

Comment:
Can we add a docs comment to explain this at a high level?

Reply:
Added 👍